lib/Frontend/Frontend.cpp

```diff
@@ -892,11 +883,9 @@ void CompilerInstance::forEachFileToTypeCheck(
 }
 
 void CompilerInstance::finishTypeChecking() {
-  if (getASTContext().TypeCheckerOpts.DelayWholeModuleChecking) {
-    forEachSourceFileIn(MainModule, [&](SourceFile &SF) {
-      performWholeModuleTypeChecking(SF);
-    });
-  }
+  forEachSourceFileIn(MainModule, [&](SourceFile &SF) {
```

Comment:
I *think* this always should have been `forEachFileToTypeCheck`. Gonna give that a shot locally. Would certainly be a win in incremental mode.